odoo/osv/expression.py

```diff
@@ -888,7 +888,10 @@ def child_of_domain(left, ids, left_model, parent=None, prefix=''):
                     records = records.search([(parent_name, 'in', records.ids)], order='id') - records.browse(child_ids)
                 domain = [('id', 'in', list(child_ids))]
             if prefix:
-                return [(left, 'in', left_model._search(domain))]
+                # We need to use sudo() here in order the retrieve the full domain
+                # including inaccessible children. Forbidden records will be filtered
+                # by the rest of the domain.
+                return [(left, 'in', left_model.sudo()._search(domain))]
```

Comment:
I would put the sudo on line 871, for the same reasons `active_test` is put there.
It is already sudoed everywhere anyways...

Reply:
Hmm, there are still a lot of accesses where sudo is not needed, so it seems a bit overkill to have everything sudo'ed. (5 without sudo vs. 3 with sudo)

Reply:
Mmmh, I'll let the ORM/security team review this anyway, but I don't see any other call to a function that can be impacted by the environment where `sudo` is not used.
The other uses are only static properties of the class where the env is not even accessible

Reply:
Fair enough. I moved the `sudo` to the top as suggested.

Reply:
@xmo-odoo , semantically I am not against at this change. It is more a security issue: now the user can indirectly (via many2one) read `parent_path` of forbidden model/record (same as for `display_name`). IMO, it's not a sensitive information, what do you think ?

If the only reason of this change is for `check_company_domain_parent_of`, maybe we can change it instead ?

Reply:
> If the only reason of this change is for `check_company_domain_parent_of`, maybe we can change it instead ?

@ryv-odoo Hmm, that's actually an interesting idea indeed. Looking into that now.

Reply:
@ryv-odoo @xmo-odoo Changing only the `check_company_domain_parent_of ` would work, although I think that the change in `expression.py` might be correct when looking at the adapted test in `base`: `test_expression.py`.

There we can't access the middle `res.partner` in the parent-chain, but when searching `res.partner.bank` records where we should have access to all three, I have the impression we should still allow searching the bank of the middle partner. What do you think?

Reply:
@ryv-odoo Can you check my previous comment to see whether we leave it like this or I change it to only modify `check_company_domain_parent_of `? Thanks.

Reply:
Agree, it seems logic to me. But the question is more for the security team: is it ok to indirectly leak parent_path of inaccessible record via one Many2one